starter-code/src/Contacts.jsx

```diff
@@ -0,0 +1,74 @@
+import React, {Component} from 'react';
+import {getFirstFive} from './contactsList';
+import {getContacts} from './contactsList';
+
+class Contacts extends Component {
+  state = {
+    contacts: getFirstFive (),
+    fulllist: getContacts()
+  };
+
+  handleDelete = contact => {
+    //let contacts = [...this.state.contacts];
+    let contacts = this.state.contacts.filter (c =>  c.name !== contact.name);
+    this.setState ({contacts});  
+  };
+
+  handleAddRandom = () => {
+    const addOne = this.state.fulllist[Math.floor(Math.random()*this.state.fulllist.length)]
+    contacts: this.state.contacts.push(addOne)
+    this.setState({})
+  }
+
+  handleSortName = () => {
+    let contacts = this.state.contacts.sort( (a,b) => {
+      return a.name < b.name ? -1 : a.name > b.name ? 1 :0
+  })
+    this.setState({contacts})
+  };
+
+
+  handlePopular = () => {
+    let contacts = this.state.contacts.sort((a,b) => {
+     return a.popularity > b.popularity ? -1 : a.popularity < b.popularity ? 1 : 0
+  })
+    this.setState({contacts})
+  }
+
+
+  render () {
```

Comment:
This component is to big. Try to split it up in smaller components, for example in the following components: 
- TableControls />
- ContactRow />